src/include/task/task.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "optimizer/cost_model/abstract_cost_model.h"
+#include "parser/expression/constant_value_expression.h"
+
+#include <vector>
+
+namespace noisepage::util {
+class QueryExecUtil;
+}
+
+namespace noisepage::task {
+
+class TaskManager;
+
+enum class TaskType : uint8_t { DDL_TASK, DML_TASK };
+
+class Task {
+ public:
+  virtual void Execute(common::ManagedPointer<util::QueryExecUtil> query_exec_util,
+                       common::ManagedPointer<task::TaskManager> task_manager) = 0;
+  virtual TaskType GetTaskType() = 0;
+
+  virtual ~Task() = default;
+};
+
+class TaskDDL : public Task {
+ public:
+  TaskDDL(catalog::db_oid_t db_oid, std::string query_text) : db_oid_(db_oid), query_text_(query_text) {}
+
+  void Execute(common::ManagedPointer<util::QueryExecUtil> query_exec_util,
+               common::ManagedPointer<task::TaskManager> task_manager) override;
+  TaskType GetTaskType() override { return TaskType::DDL_TASK; }
+
+  ~TaskDDL() override = default;
+
+ private:
+  catalog::db_oid_t db_oid_;
+  std::string query_text_;
+};
+
+class TaskDML : public Task {
+ public:
+  TaskDML(catalog::db_oid_t db_oid, std::string query_text, std::unique_ptr<optimizer::AbstractCostModel> cost_model,
+          std::vector<std::vector<parser::ConstantValueExpression>> &&params, std::vector<type::TypeId> &&param_types)
```

Comment:
Is it by design that DML tasks that go through the `TaskManager` do not have any callback functions to process the tuples?